### ISSUE TITLE
On MAC: On the same call, when providing the update and arguments to download the video, the download fails

### ISSUE THREAD

**Issue by tormyvancool** [yt-dlp/yt-dlp#11445]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

by calling yt-dlp in mac, I got this issue and upfront to open a bug report, i wanna be sure.

This happens only on MAC. On Windows and Linux it works fine.
On the same call, when providing the update and arguments to download the video, the download fails
example

`yt-dlp --update-to master -S vcodec:h264,res,acodec:aac <URL_of_the_video> -P "F:/Videos/" -o "My Title.mp4"`

- it checks for the update
- if founded: 
--    it updates itself then
--    it stops without re-run itself and going ahead to execute the arguments given to download the video.
- if not founded:
--   it goes ahead normally by downloading the video as requested

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

**Comment by bashonly** [comment]
please provide verbose logs

**Comment by tormyvancool** [comment]
> please provide verbose logs

There is nothing on the verbose log about this. Just it tells you it updates itself and I have to relaunch it :-/

**Comment by tormyvancool** [comment]
this is the first try ... when the script ends up

```
yt-dlp cd "/Users/ben/Library/Application Support/REAPER/Scripts/Tormy Van Cool ReaPack Scripts/Video/yt-dlp/" && chmod +x yt-dlp_macos && ./yt-dlp_macos --update-to master -S vcodec:h264,res,acodec:aac https://www.youtube.com/watch\?v\=9pZUokleIf0 -P "/Users/ben/poub/poub_ytdlp/Media/Videos/" -o "dsfgsdf.mp4" -v
[debug] Command-line config: ['--update-to', 'master', '-S', 'vcodec:h264,res,acodec:aac', 'https://www.youtube.com/watch?v=9pZUokleIf0', '-P', '/Users/ben/poub/poub_ytdlp/Media/Videos/', '-o', 'dsfgsdf.mp4', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.11.04 from yt-dlp/yt-dlp [197d0b03b] (darwin_exe)
[debug] Python 3.12.7 (CPython x86_64 64bit) - macOS-13.6.3-x86_64-i386-64bit-PE (OpenSSL 3.0.15 3 Sep 2024)
[debug] exe versions: ffmpeg 7.0.1 (setts), ffprobe 7.0.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.21.0, brotli-1.1.0, certifi-2024.08.30, curl_cffi-0.7.1, mutagen-1.47.0, requests-2.32.3, sqlite3-3.45.3, urllib3-2.2.3, websockets-13.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1838 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-master-builds/releases/latest
[debug] Downloading _update_spec from https://github.com/yt-dlp/yt-dlp-master-builds/releases/latest/download/_update_spec
[debug] Downloading SHA2-256SUMS from https://github.com/yt-dlp/yt-dlp-master-builds/releases/download/2024.11.04.003339/SHA2-256SUMS
Current version: stable@2024.11.04 from yt-dlp/yt-dlp
Latest version: master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds
Current Build Hash: 14829d7aea733e991b3ff8d098dbf5ec3bf5825b6fc6bf62edee62362c9704e7
Updating to master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds ...
[debug] Downloading yt-dlp_macos from https://github.com/yt-dlp/yt-dlp-master-builds/releases/download/2024.11.04.003339/yt-dlp_macos
Updated yt-dlp to master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds
[debug] Restarting: ./yt-dlp_macos --update-to master -S vcodec:h264,res,acodec:aac 'https://www.youtube.com/watch?v=9pZUokleIf0' -P /Users/ben/poub/poub_ytdlp/Media/Videos/ -o dsfgsdf.mp4 -v
```


This is the second try

```
yt-dlp cd "/Users/ben/Library/Application Support/REAPER/Scripts/Tormy Van Cool ReaPack Scripts/Video/yt-dlp/" && chmod +x yt-dlp_macos && ./yt-dlp_macos --update-to master -S vcodec:h264,res,acodec:aac https://www.youtube.com/watch\?v\=9pZUokleIf0 -P "/Users/ben/poub/poub_ytdlp/Media/Videos/" -o "dsfgsdf.mp4" -v
[debug] Command-line config: ['--update-to', 'master', '-S', 'vcodec:h264,res,acodec:aac', 'https://www.youtube.com/watch?v=9pZUokleIf0', '-P', '/Users/ben/poub/poub_ytdlp/Media/Videos/', '-o', 'dsfgsdf.mp4', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds [197d0b03b] (darwin_exe)
[debug] Python 3.12.7 (CPython x86_64 64bit) - macOS-13.6.3-x86_64-i386-64bit-PE (OpenSSL 3.0.15 3 Sep 2024)
[debug] exe versions: ffmpeg 7.0.1 (setts), ffprobe 7.0.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.21.0, brotli-1.1.0, certifi-2024.08.30, curl_cffi-0.7.1, mutagen-1.47.0, requests-2.32.3, sqlite3-3.45.3, urllib3-2.2.3, websockets-13.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1838 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-master-builds/releases/latest
Latest version: master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds
yt-dlp is up to date (master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds)
[youtube] Extracting URL: https://www.youtube.com/watch?v=9pZUokleIf0
[youtube] 9pZUokleIf0: Downloading webpage
[youtube] 9pZUokleIf0: Downloading ios player API JSON
[youtube] 9pZUokleIf0: Downloading mweb player API JSON
[youtube] 9pZUokleIf0: Downloading player 4e23410d
[debug] Saving youtube-nsig.4e23410d to cache
[debug] [youtube] Decrypted nsig xCLgIaPyBFC_r89PA => UtZwMaM_E-W0Ew
[debug] Loading youtube-nsig.4e23410d from cache
[debug] [youtube] Decrypted nsig _ltw2YPp-8-U7MhIH => wb9LVTGSpO1J4A
[youtube] 9pZUokleIf0: Downloading m3u8 information
[debug] Sort order given by user: vcodec:h264, res, acodec:aac
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, vcodec:h264(7), res, acodec:aac(10), quality, fps, hdr:12(7), source, channels, lang, proto, size, br, asr, vext, aext, hasaud, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 9pZUokleIf0: Downloading 1 format(s): 137+140
[debug] Invoking http downloader on "https://rr5---sn-25ge7nzk.googlevideo.com/videoplayback?expire=1730738140&ei=fKMoZ5OiG5OLvdIP14HO8Qg&ip=2a01%3Ae34%3Aecb1%3A6610%3A6dca%3A3d2a%3Ad5d6%3A9d51&id=o-ABXtgNxRNZm_cyj6chvtRR4pSdVLPi1pQPWeqh5AAkIf&itag=137&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&met=1730716540%2C&mh=Qv&mm=31%2C29&mn=sn-25ge7nzk%2Csn-25glenl6&ms=au%2Crdu&mv=m&mvi=5&pl=41&rms=au%2Cau&initcwndbps=1908750&vprv=1&svpuc=1&mime=video%2Fmp4&rqh=1&gir=yes&clen=11919175&dur=22.040&lmt=1726410360900648&mt=1730716152&fvip=2&keepalive=yes&fexp=51312688%2C51326932&c=IOS&txp=6308224&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cvprv%2Csvpuc%2Cmime%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRgIhANQLz3qh3PY2q-VeHk9RJhXcnE2oY0EKw097tG8bVOUsAiEAg743lsLRsQ1cpgOj8ZOZo5qBedNU_5RSYpGfYskSLtI%3D&lsparams=met%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Crms%2Cinitcwndbps&lsig=ACJ0pHgwRgIhAIX6fV3vjGebMMFVIJQhZda6pLQe1pnEmocPyehhQGGrAiEAzHSfoL3-Oh37DYl0VazyFGr4bfoRY5Vr-ZAvjH9qRd4%3D"
[download] Destination: /Users/ben/poub/poub_ytdlp/Media/Videos/dsfgsdf.f137.mp4
[download] 100% of   11.37MiB in 00:00:02 at 4.23MiB/s
[debug] Invoking http downloader on "https://rr5---sn-25ge7nzk.googlevideo.com/videoplayback?expire=1730738140&ei=fKMoZ5OiG5OLvdIP14HO8Qg&ip=2a01%3Ae34%3Aecb1%3A6610%3A6dca%3A3d2a%3Ad5d6%3A9d51&id=o-ABXtgNxRNZm_cyj6chvtRR4pSdVLPi1pQPWeqh5AAkIf&itag=140&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&met=1730716540%2C&mh=Qv&mm=31%2C29&mn=sn-25ge7nzk%2Csn-25glenl6&ms=au%2Crdu&mv=m&mvi=5&pl=41&rms=au%2Cau&initcwndbps=1908750&vprv=1&svpuc=1&mime=audio%2Fmp4&rqh=1&gir=yes&clen=358597&dur=22.105&lmt=1726409873852492&mt=1730716152&fvip=2&keepalive=yes&fexp=51312688%2C51326932&c=IOS&txp=6308224&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cvprv%2Csvpuc%2Cmime%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRQIhAO2f7wXntyuRSxUHhgUGC7XH7QGnx1-HIn_NCcQ6Feu2AiBhwEXFnT9uMrWI65w-3TaCaEG7TZ-mnyuEvYFVfvL_RQ%3D%3D&lsparams=met%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Crms%2Cinitcwndbps&lsig=ACJ0pHgwRgIhAIX6fV3vjGebMMFVIJQhZda6pLQe1pnEmocPyehhQGGrAiEAzHSfoL3-Oh37DYl0VazyFGr4bfoRY5Vr-ZAvjH9qRd4%3D"
[download] Destination: /Users/ben/poub/poub_ytdlp/Media/Videos/dsfgsdf.f140.m4a
[download] 100% of  350.19KiB in 00:00:00 at 3.49MiB/s
[Merger] Merging formats into "/Users/ben/poub/poub_ytdlp/Media/Videos/dsfgsdf.mp4"
[debug] ffmpeg command line: ffmpeg -y -loglevel repeat+info -i file:/Users/ben/poub/poub_ytdlp/Media/Videos/dsfgsdf.f137.mp4 -i file:/Users/ben/poub/poub_ytdlp/Media/Videos/dsfgsdf.f140.m4a -c copy -map 0:v:0 -map 1:a:0 -movflags +faststart file:/Users/ben/poub/poub_ytdlp/Media/Videos/dsfgsdf.temp.mp4
Deleting original file /Users/ben/poub/poub_ytdlp/Media/Videos/dsfgsdf.f140.m4a (pass -k to keep)
Deleting original file /Users/ben/poub/poub_ytdlp/Media/Videos/dsfgsdf.f137.mp4 (pass -k to keep)
```

**Comment by tormyvancool** [comment]
Please can you update form "incomplete" to 'question' since it still a question I have and it's not cleared out for the time being? Or should I open directly a Bug report?

**Comment by bashonly** [comment]
I fail to see what the problem is with those logs. It seems to be working as expected

**Comment by bashonly** [comment]
it fails to restart on its own?

**Comment by tormyvancool** [comment]
> I fail to see what the problem is with those logs. It seems to be working as expected

In fact it doesn't ... the reason is:
I have to call ti 2 times (same arguments) to have the whole action done .. this is occurring only with MAC.

Instead of in windows and linux, I call it only 1 time and it performs the whole thing

**Comment by tormyvancool** [comment]
> it fails to restart on its own?

yes exaclty!

**Comment by bashonly** [comment]
are you running an external script? or running that command in the shell directly?

**Comment by tormyvancool** [comment]
> are you running an external script? or running that command in the shell directly?

Both;
I mainly running it from LUA by calling `os.exec()`
But having strange behaviour i tired to call it form the shell and I discovered that behaviour. Hence I opened this request

**Comment by tormyvancool** [comment]
> but the issue is presenting when running yt-dlp from the shell, not from using a script?

also from the script. Which was triggering a debug action by me and I discovered that behaviour.
That's why I called the verbose reports: "first try" and "second try" ... I had to start it 2 times form the shell as well.

So: in both cases it beaves as described.

**Comment by seproDev** [comment]
I can not reproduce on macOS 15. In the log you posted your run `yt-dlp` first. Is that some kind of a wrapper script?
```
yt-dlp cd "/Users/...
```
I tested being in a different folder and the running
```
➜  /Library cd "/Users/username/Downloads/" && chmod +x yt-dlp_macos && ./yt-dlp_macos --update-to master https://www.youtube.com/watch\?v\=9pZUokleIf0 -v --simulate
[debug] Command-line config: ['--update-to', 'master', 'https://www.youtube.com/watch?v=9pZUokleIf0', '-v', '--simulate']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.11.04 from yt-dlp/yt-dlp [197d0b03b] (darwin_exe)
[debug] Python 3.12.7 (CPython arm64 64bit) - macOS-15.1-arm64-arm-64bit (OpenSSL 3.0.15 3 Sep 2024)
[debug] exe versions: ffmpeg 7.1 (setts), ffprobe 7.1
[debug] Optional libraries: Cryptodome-3.21.0, brotli-1.1.0, certifi-2024.08.30, curl_cffi-0.7.1, mutagen-1.47.0, requests-2.32.3, sqlite3-3.45.3, urllib3-2.2.3, websockets-13.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Extractor Plugins: VerifyImpersonateIE
[debug] Plugin directories: ['/Users/username/Downloads/yt-dlp-plugins/yt-dlp-verifyimpersonate/yt_dlp_plugins']
[debug] Loaded 1839 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-master-builds/releases/latest
[debug] Downloading _update_spec from https://github.com/yt-dlp/yt-dlp-master-builds/releases/latest/download/_update_spec
[debug] Downloading SHA2-256SUMS from https://github.com/yt-dlp/yt-dlp-master-builds/releases/download/2024.11.04.003339/SHA2-256SUMS
Current version: stable@2024.11.04 from yt-dlp/yt-dlp
Latest version: master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds
Current Build Hash: 14829d7aea733e991b3ff8d098dbf5ec3bf5825b6fc6bf62edee62362c9704e7
Updating to master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds ...
[debug] Downloading yt-dlp_macos from https://github.com/yt-dlp/yt-dlp-master-builds/releases/download/2024.11.04.003339/yt-dlp_macos
Updated yt-dlp to master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds
[debug] Restarting: ./yt-dlp_macos --update-to master 'https://www.youtube.com/watch?v=9pZUokleIf0' -v --simulate
[debug] Command-line config: ['--update-to', 'master', 'https://www.youtube.com/watch?v=9pZUokleIf0', '-v', '--simulate']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds [197d0b03b] (darwin_exe)
[debug] Python 3.12.7 (CPython arm64 64bit) - macOS-15.1-arm64-arm-64bit (OpenSSL 3.0.15 3 Sep 2024)
[debug] exe versions: ffmpeg 7.1 (setts), ffprobe 7.1
[debug] Optional libraries: Cryptodome-3.21.0, brotli-1.1.0, certifi-2024.08.30, curl_cffi-0.7.1, mutagen-1.47.0, requests-2.32.3, sqlite3-3.45.3, urllib3-2.2.3, websockets-13.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Extractor Plugins: VerifyImpersonateIE
[debug] Plugin directories: ['/Users/username/Downloads/yt-dlp-plugins/yt-dlp-verifyimpersonate/yt_dlp_plugins']
[debug] Loaded 1839 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-master-builds/releases/latest
Latest version: master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds
yt-dlp is up to date (master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds)
[youtube] Extracting URL: https://www.youtube.com/watch?v=9pZUokleIf0
[youtube] 9pZUokleIf0: Downloading webpage
[youtube] 9pZUokleIf0: Downloading ios player API JSON
[youtube] 9pZUokleIf0: Downloading mweb player API JSON
[debug] Loading youtube-nsig.4e23410d from cache
[debug] [youtube] Decrypted nsig TfuS_m2_arWw5TU9L => PR5_LWc9USxuyQ
[debug] Loading youtube-nsig.4e23410d from cache
[debug] [youtube] Decrypted nsig rkulYFWYg8POYxv37 => 1jq6mlBjUO6Aug
[youtube] 9pZUokleIf0: Downloading m3u8 information
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec, channels, acodec, lang, proto, size, br, asr, vext, aext, hasaud, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 9pZUokleIf0: Downloading 1 format(s): 313+251
```

**Comment by tormyvancool** [comment]
> yt-dlp cd "/Users/...

it's the command into the script.

**Comment by tormyvancool** [comment]
> I have asked you multiple times if you were using a script or calling yt-dlp directly from the shell. Those logs were with a script? Please USE YT-DLP DIRECTLY. We are not here to support your scripts

I also told you several times that the final result do not change because I got the issue in both cases.

however that log was obtained by encap the command line into os.execute() (LUA)

Here launched in the shell .. as you can see it fails to restart itself.

```
yt-dlp ./yt-dlp_macos --update-to master -S vcodec:h264,res,acodec:aac "https://www.youtube.com/watch\?v\=9pZUokleIf0" -P "/Users/ben/Downloads" -o "dsfgsdf.mp4" -v
[debug] Command-line config: ['--update-to', 'master', '-S', 'vcodec:h264,res,acodec:aac', 'https://www.youtube.com/watch\\?v\\=9pZUokleIf0', '-P', '/Users/ben/Downloads', '-o', 'dsfgsdf.mp4', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.11.04 from yt-dlp/yt-dlp [197d0b03b] (darwin_exe)
[debug] Python 3.12.7 (CPython x86_64 64bit) - macOS-13.6.3-x86_64-i386-64bit-PE (OpenSSL 3.0.15 3 Sep 2024)
[debug] exe versions: ffmpeg 7.0.1 (setts), ffprobe 7.0.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.21.0, brotli-1.1.0, certifi-2024.08.30, curl_cffi-0.7.1, mutagen-1.47.0, requests-2.32.3, sqlite3-3.45.3, urllib3-2.2.3, websockets-13.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1838 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-master-builds/releases/latest
[debug] Downloading _update_spec from https://github.com/yt-dlp/yt-dlp-master-builds/releases/latest/download/_update_spec
[debug] Downloading SHA2-256SUMS from https://github.com/yt-dlp/yt-dlp-master-builds/releases/download/2024.11.04.003339/SHA2-256SUMS
Current version: stable@2024.11.04 from yt-dlp/yt-dlp
Latest version: master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds
Current Build Hash: 14829d7aea733e991b3ff8d098dbf5ec3bf5825b6fc6bf62edee62362c9704e7
Updating to master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds ...
[debug] Downloading yt-dlp_macos from https://github.com/yt-dlp/yt-dlp-master-builds/releases/download/2024.11.04.003339/yt-dlp_macos
Updated yt-dlp to master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds
[debug] Restarting: ./yt-dlp_macos --update-to master -S vcodec:h264,res,acodec:aac 'https://www.youtube.com/watch\?v\=9pZUokleIf0' -P /Users/ben/Downloads -o dsfgsdf.mp4 -v
```

**Comment by bashonly** [comment]
> ```
> yt-dlp ./yt-dlp_macos --update-to master -S vcodec:h264,res,acodec:aac "https://www.youtube.com/watch\?v\=9pZUokleIf0" -P "/Users/ben/Downloads" -o "dsfgsdf.mp4" -v
> ```

what is this though? why are you running `yt-dlp ./yt-dlp_macos ...`? it looks like a wrapper script or something.

**Comment by BenTalagan** [comment]
Hi I'm jumping in the discussion because it's not productive. What Tormy is actually posting are the results of a common debugging session, he's not working under macos but asks me to launch some commands and as posted the results here so the information is not formatted correctly and hardly understandable. 

Regarding the `yt-dlp ./yt-dlp_macos` stuff, the first part is just the prompt of my shell (it was the name of the folder where the yt-dlp_macos binary was located).

Please, make abstraction of what's been posted precedently and follow these simple steps. From a macos terminal : 

- Download yt-dlp_macos from the github release page
- Go to the Downloads directory (or the one where you stored the binary)
- chmod +x yt-dlp_macos
- Launch the following command : 

```./yt-dlp_macos --update-to master -S vcodec:h264,res,acodec:aac "https://www.youtube.com/watch?v=9pZUokleIf0" -o "test.mp4" -v```

(at this point macos asks for authorizations, that you should give from the preferences panel)

Re-launch the command. You will get the following output : 

```
[debug] Command-line config: ['--update-to', 'master', '-S', 'vcodec:h264,res,acodec:aac', 'https://www.youtube.com/watch?v=9pZUokleIf0', '-o', 'test.mp4', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.11.04 from yt-dlp/yt-dlp [197d0b03b] (darwin_exe)
[debug] Python 3.12.7 (CPython x86_64 64bit) - macOS-13.6.3-x86_64-i386-64bit (OpenSSL 3.0.15 3 Sep 2024)
[debug] exe versions: ffmpeg 7.0.1 (setts), ffprobe 7.0.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.21.0, brotli-1.1.0, certifi-2024.08.30, curl_cffi-0.7.1, mutagen-1.47.0, requests-2.32.3, sqlite3-3.45.3, urllib3-2.2.3, websockets-13.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1838 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-master-builds/releases/latest
[debug] Downloading _update_spec from https://github.com/yt-dlp/yt-dlp-master-builds/releases/latest/download/_update_spec
[debug] Downloading SHA2-256SUMS from https://github.com/yt-dlp/yt-dlp-master-builds/releases/download/2024.11.04.003339/SHA2-256SUMS
Current version: stable@2024.11.04 from yt-dlp/yt-dlp
Latest version: master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds
Current Build Hash: 14829d7aea733e991b3ff8d098dbf5ec3bf5825b6fc6bf62edee62362c9704e7
Updating to master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds ...
[debug] Downloading yt-dlp_macos from https://github.com/yt-dlp/yt-dlp-master-builds/releases/download/2024.11.04.003339/yt-dlp_macos
Updated yt-dlp to master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds
[debug] Restarting: ./yt-dlp_macos --update-to master -S vcodec:h264,res,acodec:aac 'https://www.youtube.com/watch?v=9pZUokleIf0' -o test.mp4 -v
```

It stops here, yt-dlp_macos is now up to date, but the restart is not effective. The command needs to be launched again manually to launch the download, while yt-dlp_macos says it's restarting.

Basically, that's what Tormy is complaining about.

**Comment by seproDev** [comment]
Thank you for the explanation. However even with these steps I can not replicate on my machine. I also do not get asked for authorization by macos. Potentially this is some security mechanism by Apple, but until another dev can reproduce this will be difficult to debug and fix.

**Comment by tormyvancool** [comment]
Thank you @BenTalagan for our intervention
@bashonly I see you changed into "cant-reproduce" but this is what' occurring not only to Ben but other users
Indeed i involved Ben because the other ones were not able to provide a more technical feedback.
My conclusion is that there is some anomaly on MAC only

Linux and Windows' versions are working fine

**Comment by bashonly** [comment]
> `cant-reproduce`: "The issue cannot be reliably reproduced"

I'd say that's accurate, since the developer commenting on this issue cannot reproduce the issue

**Comment by BenTalagan** [comment]
@seproDev regarding the authorizations, maybe you've authorized your system to launch applications coming from any developer ?

This can be checked in Settings > Privacy & Security > Allow apps from anywhere

Normally, this option cannot be enabled through the UI, it is hidden and is triggered by : 

```sudo spctl --master-disable```

Standard users generally don't have this option on, but only trust the app store and "identified developers".

Anyway, I've changed this parameter on my own machine and remade the test. It effectively disables the authorization popup (so we spare the auth step), but the rest of the problem is the same, on the first launch yt-dlp updates itself, says it's going to restart to perform the download of the video, but it's not restarting. The command needs to be launched again.

For info right now I'm testing on Ventura 13.6.3.

**Comment by seproDev** [comment]
Allow application from is set to "App Store & Known Developers" on my machine. I am on macOS Sequoia 15.1.

There have been occasional reports of insanely long startup times on macOS, likely due to gatekeeper. I was never able to reproduce these either.

**Comment by BenTalagan** [comment]
Indeed, it takes about 30 seconds to yt-dlp_macos on my machine to output the first log, while when launching the brew version, it takes less than 10 seconds.

**Comment by seproDev** [comment]
On my machine both the pyinstaller binary and the brew version take about a second for the first log line.

I fear that signing of the yt-dlp_macos binary might be required, at least for the long startup times, but I am not super familiar with macos development and it's security architecture.
For the issue at hand, I could also see this being a pyinstaller regression. Do older yt-dlp builds correctly restart?

**Comment by BenTalagan** [comment]
Hmm, I downloaded the very first version with the --update-to option. Same behaviour : 

```
➜  Downloads chmod +x yt-dlp_macos
➜  Downloads ./yt-dlp_macos --version
2023.03.03
➜  Downloads ./yt-dlp_macos --update-to stable@2023.06.21 -S vcodec:h264,res,acodec:aac "https://www.youtube.com/watch?v=9pZUokleIf0" -o "test.mp4" -v
[debug] Command-line config: ['--update-to', 'stable@2023.06.21', '-S', 'vcodec:h264,res,acodec:aac', 'https://www.youtube.com/watch?v=9pZUokleIf0', '-o', 'test.mp4', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2023.03.03 [934496428] (darwin_exe)
[debug] Python 3.8.9 (CPython x86_64 64bit) - macOS-13.6.3-x86_64-i386-64bit (LibreSSL 2.8.3)
[debug] exe versions: ffmpeg 7.0.1 (setts), ffprobe 7.0.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1782 extractors
Current version: stable@2023.03.03
Current Build Hash: 4b24d093196efdb0abe4968c9cca07523f648b9bd43c23c893250f8c7d1435af
[debug] Downloading _update_spec from https://github.com/yt-dlp/yt-dlp/releases/latest/download/_update_spec
Updating to stable@2023.06.21 ...
[debug] Downloading yt-dlp_macos from https://github.com/yt-dlp/yt-dlp/releases/download/2023.06.21/yt-dlp_macos
[debug] Downloading SHA2-256SUMS from https://github.com/yt-dlp/yt-dlp/releases/download/2023.06.21/SHA2-256SUMS
Updated yt-dlp to stable@2023.06.21
[debug] Restarting: ./yt-dlp_macos --update-to stable@2023.06.21 -S vcodec:h264,res,acodec:aac 'https://www.youtube.com/watch?v=9pZUokleIf0' -o test.mp4 -v
➜  Downloads ./yt-dlp_macos --version                                                                                                                 
2023.06.21
```

**Comment by pms967** [comment]
@BenTalagan As a possible workaround, you may try to wrap yt-dlp in a shell script  like this: 
`
#!/bin/sh
/path/to/yt-dlp_macos --update-to master 
/path/to/yt-dlp_macos -S vcodec:h264,res,acodec:aac "$1" -o "test.mp4" -v
`

save it as e.g. yp-dlp.sh, make it executable (chmod +x yp-dlp.sh), and try to run it passing the URL as the first (and only) argument.

I know nothing about MacOS/X oddities but, in principle, that should do the trick...

**Comment by BenTalagan** [comment]
@pms967 thanks for the suggestion ! Unfortunately, I think @tormyvancool's requirements were to be able to do everything in one call for other reasons (yt-dlp is called inside a lua script which uses os.execute and the way yt-dlp outputs the logs does trigger some problems, but here it's not my part anymore :) )

**Comment by tormyvancool** [comment]
> @pms967 thanks for the suggestion ! Unfortunately, I think @tormyvancool's requirements were to be able to do everything in one call for other reasons (yt-dlp is called inside a lua script which uses os.execute and the way yt-dlp outputs the logs does trigger some problems, but here it's not my part anymore :) )

Exactly and this I also explained to @pms967 very well. The multiple call is out of any option right for what you stated, since it was the very first opton adopted btu then abandoned. Too many troubles generated and I won't be back on my decision

**Comment by pms967** [comment]
> @pms967 thanks for the suggestion ! Unfortunately, I think @tormyvancool's requirements were to be able to do everything in one call for other reasons (yt-dlp is called inside a lua script which uses os.execute and the way yt-dlp outputs the logs does trigger some problems, but here it's not my part anymore :) )

That's not a problem at all, since he will have to call the script (once) and not the yt-dlp executable.

All he have to do is add the script to the files being downloaded, make it executable and run it (from the LUA script).

**Comment by tormyvancool** [comment]
> > @pms967 thanks for the suggestion ! Unfortunately, I think @tormyvancool's requirements were to be able to do everything in one call for other reasons (yt-dlp is called inside a lua script which uses os.execute and the way yt-dlp outputs the logs does trigger some problems, but here it's not my part anymore :) )
> 
> That's not a problem at all, since he will have to call the script (once) and not the yt-dlp executable.
> 
> All he have to do is to add the script to the files being downloaded, make it executable and run it (from the LUA script).
> 
> But, first, you should check if running the script manually from the shell will work as expected.

No this IS a problem. A huge problem! I really well explained it to you in messenger.
To call the way you suggested I have to invoke os.Execute()  for each line you wrote.
This is NOT doable because every time yt-dlp finishes any operation, returns a "done" to the calling function even if the entire process is NOT done. So the LUA script will go on, but yt-dlp have not finished its operations.

so this suggestion is definitely dropped and I no want to reconsider it any longer. Too many useless troubles.

Look:
On WIndows is working fine by calling it 1 time.
On Linux the same
It must work fine as it is already for MAC. This is the line to keep.

**Comment by pms967** [comment]
> No this IS a problem. A huge problem! I really well explained it to you in messenger. To call the way you suggested I have to invoke os.Execute() for each line you wrote. 

No!

you only have to `os.Execute("chmod +x /path/to/yp-dlp.sh ; path/to/yp-dlp.sh "https://www.youtube.com/watch?v=9pZUokleIf0"")`

and that's it!

>This is NOT doable because every time yt-dlp finishes any operation, returns a "done" to the calling function even if the entire process is NOT done. So the LUA script will go on, but yt-dlp have not finished its operations.

Unless MacOS/X is completely broken, that doesn't matter. The system is running the SHELL SCRIPT, and will have to wait for its completion. Not that of whatever command / executable is run from within a script.

**Comment by tormyvancool** [comment]
> > No this IS a problem. A huge problem! I really well explained it to you in messenger. To call the way you suggested I have to invoke os.Execute() for each line you wrote.
> 
> No!
> 
> you only have to `os.Execute('chmod +x /path/to/yp-dlp.sh ; path/to/yp-dlp.sh "https://www.youtube.com/watch?v=9pZUokleIf0"')`
> 
> and that's it!
> 
> > This is NOT doable because every time yt-dlp finishes any operation, returns a "done" to the calling function even if the entire process is NOT done. So the LUA script will go on, but yt-dlp have not finished its operations.
> 
> Unless MacOS/X is completely broken, that doesn't matter. The system is running the SHELL SCRIPT, and will have to wait for its completion. Not that of whatever command / executable is run from within a script.

This call is what it does already and you missed the update
Those are the requirements and i will not change them. Ok?
So:
- Each time the lua script starts, has to update yt-dlp to the very last version  if available
- Once finished it has to call itself and to download the video
- then the script goes ahead with LUA functions that are not involving yt-dlp any longer

The correct line is this one
`yt-dlp --update-to master -S vcodec:h264,res,acodec:aac <URL_of_the_video> -P "F:/Videos/" -o "My Title.mp4"`

The calls to ` chmod +x `etc, are done before the above line is called. Indeed yt-dlp is executed.

**Comment by pms967** [comment]
@seproDev After discussing in private with Tormy (to avoid polluting here...), it looks like  at the bottom there is (also?) a more severe problem, which does not affect only MacOS. 

According to his tests - in Windows - it looks like  the control returns to the calling LUA script as soon as yt-dlp write something "on screen", thus obviously before it has completed whatever operation(s) it was instructed to do.

But, at least from the command line (on Linux), I've been unable to replicate such weird behavior.  

@tormyvancool Could it be that the LUA os.execute() function call launch the specified command(s) in background, thus returning immediately?  :-/

Maybe to get the LUA script to wait for exit you have to ask for the exit code (even if you don't use it)?
```local exit_code = os.execute("your_command")```

(just guessing... but it's worth trying, just in case. The LUA documentation I've been able to find is really scarce, and doesn't say anything about the blocking or non-blocking behavior of the os.execute function)

**Comment by BenTalagan** [comment]
Hi @pms967, to give you some more background we're coming from the REAPER D.A.W. community, which embeds its own version of a lua interpreter. It is not the system standard interpreter, but it is compiled with REAPER and executed from within REAPER's process. So it should be noted that this particular ecosystem might affect the behaviour of os.execute. I can perform a few tests under macos to give you some more background if you want.

**Comment by BenTalagan** [comment]
First test, getting the version : 

From macos lua interpreter : 

```
➜  Downloads $ lua
Lua 5.4.7  Copyright (C) 1994-2024 Lua.org, PUC-Rio
> ret, one, two, three = os.execute("/Users/ben/Downloads/yt-dlp_macos --version")
2024.11.04
> ret
true
> one
exit
> two
0
> three
nil
> 
```

From REAPER's interpreter, using the following script : 

```lua
 ret, one, two, three = os.execute("/Users/ben/Downloads/yt-dlp_macos --version")

function to_s(v)
  if v == true then return "true" end
  if v == false then return "false" end
  if v == nil then return "nil" end
  return "" .. v
end

reaper.ShowConsoleMsg("" .. to_s(ret) .. "\n");
reaper.ShowConsoleMsg("" .. to_s(one) .. "\n");
reaper.ShowConsoleMsg("" .. to_s(two) .. "\n");
reaper.ShowConsoleMsg("" .. to_s(three) .. "\n");
```

The output is the following : 

```
true
exit
0
nil
```

And is coherent.

**Comment by BenTalagan** [comment]
From macos lua interpreter : 

```
➜  Downloads $ lua
Lua 5.4.7  Copyright (C) 1994-2024 Lua.org, PUC-Rio
> ret, one, two, three = os.execute('./yt-dlp_macos --update-to master "https://www.youtube.com/watch?v=9pZUokleIf0" -o "test.mp4" -v')
[debug] Command-line config: ['--update-to', 'master', 'https://www.youtube.com/watch?v=9pZUokleIf0', '-o', 'test.mp4', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.11.04 from yt-dlp/yt-dlp [197d0b03b] (darwin_exe)
[debug] Python 3.12.7 (CPython x86_64 64bit) - macOS-13.6.8-x86_64-i386-64bit (OpenSSL 3.0.15 3 Sep 2024)
[debug] exe versions: ffmpeg 7.0.2 (setts), ffprobe 7.0.2
[debug] Optional libraries: Cryptodome-3.21.0, brotli-1.1.0, certifi-2024.08.30, curl_cffi-0.7.1, mutagen-1.47.0, requests-2.32.3, sqlite3-3.45.3, urllib3-2.2.3, websockets-13.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1838 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-master-builds/releases/latest
[debug] Downloading _update_spec from https://github.com/yt-dlp/yt-dlp-master-builds/releases/latest/download/_update_spec
[debug] Downloading SHA2-256SUMS from https://github.com/yt-dlp/yt-dlp-master-builds/releases/download/2024.11.04.003339/SHA2-256SUMS
Current version: stable@2024.11.04 from yt-dlp/yt-dlp
Latest version: master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds
Current Build Hash: 14829d7aea733e991b3ff8d098dbf5ec3bf5825b6fc6bf62edee62362c9704e7
Updating to master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds ...
[debug] Downloading yt-dlp_macos from https://github.com/yt-dlp/yt-dlp-master-builds/releases/download/2024.11.04.003339/yt-dlp_macos
Updated yt-dlp to master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds
[debug] Restarting: ./yt-dlp_macos --update-to master 'https://www.youtube.com/watch?v=9pZUokleIf0' -o test.mp4 -v
> ret
nil
> one
exit
> two
247
> three
nil
> 
```

From REAPER : 

```
nil
exit
247
nil
```

So the result are coherent too.

**Comment by BenTalagan** [comment]
Finally, on second try (yt-dlp is already up to date) : 

From macos lua interpreter : 

```
➜  Downloads $ lua
Lua 5.4.7  Copyright (C) 1994-2024 Lua.org, PUC-Rio
> ret, one, two, three = os.execute('./yt-dlp_macos --update-to master "https://www.youtube.com/watch?v=9pZUokleIf0" -o "test.mp4" -v')
[debug] Command-line config: ['--update-to', 'master', 'https://www.youtube.com/watch?v=9pZUokleIf0', '-o', 'test.mp4', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds [197d0b03b] (darwin_exe)
[debug] Python 3.12.7 (CPython x86_64 64bit) - macOS-13.6.8-x86_64-i386-64bit (OpenSSL 3.0.15 3 Sep 2024)
[debug] exe versions: ffmpeg 7.0.2 (setts), ffprobe 7.0.2
[debug] Optional libraries: Cryptodome-3.21.0, brotli-1.1.0, certifi-2024.08.30, curl_cffi-0.7.1, mutagen-1.47.0, requests-2.32.3, sqlite3-3.45.3, urllib3-2.2.3, websockets-13.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1838 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-master-builds/releases/latest
Latest version: master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds
yt-dlp is up to date (master@2024.11.04.003339 from yt-dlp/yt-dlp-master-builds)
[youtube] Extracting URL: https://www.youtube.com/watch?v=9pZUokleIf0
[youtube] 9pZUokleIf0: Downloading webpage
[youtube] 9pZUokleIf0: Downloading ios player API JSON
[youtube] 9pZUokleIf0: Downloading mweb player API JSON
[debug] Loading youtube-nsig.4e23410d from cache
[debug] [youtube] Decrypted nsig Lky868EM_JCX0Rahq => Y2Bii05n97PdiA
[debug] Loading youtube-nsig.4e23410d from cache
[debug] [youtube] Decrypted nsig sYZolAqScMn33yE5d => mjPUfMDc0t-9xQ
[youtube] 9pZUokleIf0: Downloading m3u8 information
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec, channels, acodec, lang, proto, size, br, asr, vext, aext, hasaud, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 9pZUokleIf0: Downloading 1 format(s): 313+251
[debug] Invoking http downloader on "https://rr2---sn-25glenl6.googlevideo.com/videoplayback?expire=1730773562&ei=2i0pZ7nRJer0xN8P5f-voQg&ip=2a01%3Ae34%3Aecac%3A71c0%3A9140%3A7391%3A848d%3A75e6&id=o-ACjcnIYw2ocx06XHnN9E2XIC739VePKZojko9Y56GnJU&itag=313&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&met=1730751962%2C&mh=Qv&mm=31%2C29&mn=sn-25glenl6%2Csn-25ge7nzk&ms=au%2Crdu&mv=m&mvi=2&pl=41&rms=au%2Cau&initcwndbps=1451250&vprv=1&svpuc=1&mime=video%2Fwebm&rqh=1&gir=yes&clen=39962927&dur=22.040&lmt=1726410219478362&mt=1730751438&fvip=5&keepalive=yes&fexp=51312688%2C51326932&c=IOS&txp=630F224&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cvprv%2Csvpuc%2Cmime%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRQIgMwPVY8CCKIZhvdwW32QE6FV9Du0I8ZN8YQcMdBmR50ACIQDU8SItX8TOCths1Ij7PWhxkCkB5-HkxyJ_XVSE15JIyw%3D%3D&lsparams=met%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Crms%2Cinitcwndbps&lsig=ACJ0pHgwRQIgdjVxpq8ulrJDCgYBMf1OU4HZKNJG7TB7kcuiwT9DZ9cCIQDGGRbSQqep7DcF7Kmp2iwutgPu9C3RjLkmW6lpBAIbIA%3D%3D"
[download] Destination: test.mp4.f313.webm
[download] 100% of   38.11MiB in 00:00:00 at 43.57MiB/s
[debug] Invoking http downloader on "https://rr2---sn-25glenl6.googlevideo.com/videoplayback?expire=1730773562&ei=2i0pZ7nRJer0xN8P5f-voQg&ip=2a01%3Ae34%3Aecac%3A71c0%3A9140%3A7391%3A848d%3A75e6&id=o-ACjcnIYw2ocx06XHnN9E2XIC739VePKZojko9Y56GnJU&itag=251&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&met=1730751962%2C&mh=Qv&mm=31%2C29&mn=sn-25glenl6%2Csn-25ge7nzk&ms=au%2Crdu&mv=m&mvi=2&pl=41&rms=au%2Cau&initcwndbps=1451250&vprv=1&svpuc=1&mime=audio%2Fwebm&rqh=1&gir=yes&clen=370503&dur=22.081&lmt=1726409851069111&mt=1730751438&fvip=5&keepalive=yes&fexp=51312688%2C51326932&c=IOS&txp=6308224&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cvprv%2Csvpuc%2Cmime%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRQIhAIi0K6u-PKeNX_DmQvNmQKrG2caI1MXrc4NQyHGf4pvkAiBHOviNgBtsF9f6cYfFgdk6VxEJPWFT7eM9cDiS_1GPgA%3D%3D&lsparams=met%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Crms%2Cinitcwndbps&lsig=ACJ0pHgwRQIgdjVxpq8ulrJDCgYBMf1OU4HZKNJG7TB7kcuiwT9DZ9cCIQDGGRbSQqep7DcF7Kmp2iwutgPu9C3RjLkmW6lpBAIbIA%3D%3D"
[download] Destination: test.mp4.f251.webm
[download] 100% of  361.82KiB in 00:00:00 at 5.98MiB/s
[Merger] Merging formats into "test.mp4.webm"
[debug] ffmpeg command line: ffmpeg -y -loglevel repeat+info -i file:test.mp4.f313.webm -i file:test.mp4.f251.webm -c copy -map 0:v:0 -map 1:a:0 -movflags +faststart file:test.mp4.temp.webm
Deleting original file test.mp4.f251.webm (pass -k to keep)
Deleting original file test.mp4.f313.webm (pass -k to keep)
> ret
true
> one
exit
> two
0
> three
nil
> 
```

From REAPER's lua interpreter : 

```
true
exit
0
nil
```

Still coherent. So under macos, IMHO, testing the return codes is probably a safe way to achieve it.

**Comment by pms967** [comment]
@BenTalagan I know. But the main problem seems to be completely unrelated to LUA, let alone Reaper.

Tormy has tried also from the windows command line, and he says that the windows prompt comes back before yt-dlp has finished to do what it's doing.

On Linux, this does not happen (at least, to me). 

What about MacOS? 

Try manually running from the unix shell (terminal) a command like this:
`./yt-dlp_macos --update-to master -S vcodec:h264,res,acodec:aac "https://www.youtube.com/watch?v=9pZUokleIf0" -o "test.mp4" -v`

and, while it's running, hit "Enter" several times. 

Does the shell prompt come back before yt-dlp has completed its job, or you get several empty prompt lines only after the last yt-dlp message?

**Comment by BenTalagan** [comment]
Under macos, if I hit "enter" during the command, the behaviour is perfectly normal : empty lines appear as soon as I enter them but it does not affect the progression yt-dlp, which goes to the end. Then, at the end, I get back my prompt.

It's to be noted that under windows, Tormy launches yt-dlp with os.execute('start "yt-dlp .... '), and the "start" command specifically launches the progress in background so it's normal that the lua script normal continues its execution just after the os.execute call and does not wait for the return code.

**Comment by BenTalagan** [comment]
@tormyvancool Have you considered using `start /B /WAIT` under windows ? Does it make a difference ?

**Comment by Grub4K** [comment]
> But the main problem seems to be completely unrelated to LUA, let alone Reaper.

I think this is true, this discussion is derailing into something thats not related to the original issue anymore. Please keep it on track with the issue at hand, the MacOS build not restarting properly after auto-updating. The additional posts makes it hard to properly find information related to the issue.

PS: Maybe discussion on the discord or some private channel would be more appropriate for solving the lua issue. Also for windows you should not use `start` at all if possible.

---
To summarize: The MacOS build exits instead of restarting itself, which results in an error code of 247.

Current open investigation points:
- Any MacOS security policy blocking the file?
  - After running failing the command, check "System Settings/Preferences > Security & Privacy". Did any file actually try to launch (as can be seen in that list) but was blocked?
- What does exit code 247 stand for? Only thing I could find was related to OOM errors, nothing in other, related topics
- Eventually a minimal reproduction case, possibly using a simple pyinstaller python file downloaded from the internet

**Comment by BenTalagan** [comment]
To recenter on the original problem as stated by @Grub4K, one more info : the problem does not happen when using a checkout of the github project. On my machine, if I git clone the repository, cd into it, make it and launch : 

```
./yt-dlp --update-to master "https://www.youtube.com/watch?v=9pZUokleIf0" -o "test.mp4" -v
```

Then yt-dlp restarts correctly after update and starts the download, when the yt-dlp_macos binary provided in the releases doesn't.

**Comment by tormyvancool** [comment]
> /B /WAIT

Unfortunately not any difference.

**Comment by tormyvancool** [comment]
> @tormyvancool Have you considered using `start /B /WAIT` under windows ? Does it make a difference ?

I wanted to retry now after you suggested me the doc to read. It works :-) I have made some mistake yesterday while I implemented it.
My bad. My dep apologize.

Now still to solve the MAC OS issue

**Comment by tormyvancool** [comment]
@Grub4K @bashonly @seproDev guys, just to let you know that @BenTalagan have tested now once again the LUA script and yes we confirm that in MAC OS the double call is required or YT-DLP aborts as we said in the previous messages.
However it's also very slow if compared with Win version

@pms967 is it enough quick under Linux? i mean is it acceptable?

Something is weird on Mac

**Comment by bashonly** [comment]
> On my machine, if I git clone the repository, cd into it, make it and launch :
> ```
> ./yt-dlp --update-to master "https://www.youtube.com/watch?v=9pZUokleIf0" -o "test.mp4" -v
> ```
> Then yt-dlp restarts correctly after update and starts the download, when the yt-dlp_macos binary provided in the releases doesn't.

@BenTalagan When you say "make it", do you mean you're running `make` and making a zipimport build? Or are you running `bundle/pyinstaller.py` and building an actual pyinstaller executable (like what the `yt-dlp_macos` binary is)? 

> However it's also very slow if compared with Win version

@tormyvancool As mentioned [above](https://github.com/yt-dlp/yt-dlp/issues/11445#issuecomment-2454907229), this is a separate issue that is(/was) tracked in #10425. The issue is closed since it is an issue with *all* pyinstaller-bundled executables, not just yt-dlp. The pyinstaller devs have stated that it is a problem inherent to all `--onefile` executables, and most likely will never / can never be solved. There may be some sort of workaround with disabling Apple security scanning/gatekeeper, but that's probably not something we or you should advise all of our macOS users to do.

**Comment by bashonly** [comment]
One thing we could test is if the macOS executables built with PyInstaller <6.10 also fail to restart after updating. Try using https://github.com/yt-dlp/yt-dlp/releases/download/2024.08.06/yt-dlp_macos instead of latest stable as a starting point for your `./yt-dlp_macos --update-to master "URL"` test.

**Comment by pms967** [comment]
> @pms967 is it enough quick under Linux? i mean is it acceptable?

there's no perceivable delay or slowness WRT the same command run manually from the command line.

**Comment by bashonly** [comment]
> According to his tests - in Windows - it looks like the control returns to the calling LUA script as soon as yt-dlp write something "on screen", thus obviously before it has completed whatever operation(s) it was instructed to do.

This is being tracked in a separate issue as well: #11355

I asked a couple questions to try to investigate the issue but didn't get an answer for either of them

**Comment by BenTalagan** [comment]
> @BenTalagan When you say "make it", do you mean you're running make and making a zipimport build? Or are you running bundle/pyinstaller.py and building an actual pyinstaller executable (like what the yt-dlp_macos binary is)?

Just running `make` in a fresh checkout directory, and launching the produced executable (it looks like it starts with '#!/usr/bin/env python3' and then a zip made of python script is append). I don't build yt-dlp_macos (I don't know how), and the produced executable is smaller by far.

**Comment by pms967** [comment]
@BenTalagan
> Just running `make` in a fresh checkout directory, and launching the produced executable (it looks like it starts with '#!/usr/bin/env python3' and then a zip made of python script files is appended). I don't build yt-dlp_macos (I don't know how), and the produced executable is smaller by far.

Yeah, because (differently from yt-dlp_macos) this contains only the Python code of yt-dlp, and rely on a Python interpreter already installed on the system. Of course it won't run if a viable version of Python3 can't be found.

**Comment by bashonly** [comment]
@pms967 is correct, and to expand on what was said: `yt-dlp` (the zipimport executable product of `make`) does not unpack its contents to /tmp, like how `yt-dlp_macos` (the pyinstaller-bundled executable) needs to do. The needed unpacking is known to be the cause of the slowdown (again, to be discussed in #10425), and I wouldn't be surprised it's the cause of the post-update restart failure as well.

**Comment by bashonly** [comment]
Though it's also notable that the zip build uses a different code path for updating than the pyinstaller exe's do:

https://github.com/yt-dlp/yt-dlp/blob/282e19db827f0951c783ac946429f662bcf2200c/yt_dlp/update.py#L462-L521
